modules/testing/types.go

```diff
@@ -0,0 +1,18 @@
+package testing
+
+type TestingT interface {
+	Fail()
+	Error(args ...interface{})
+	Errorf(format string, args ...interface{})
+	FailNow()
+	Fatal(args ...interface{})
+	Fatalf(format string, args ...interface{})
+	Log(args ...interface{})
+	Logf(format string, args ...interface{})
+	Failed() bool
+	Parallel()
+	Skip(args ...interface{})
+	Skipf(format string, args ...interface{})
+	SkipNow()
+	Skipped() bool
```

Comment:
I suspect in Terratest we only use a couple off these, so this interface can probably be reduced quite a bit.

Reply:
This is the interface as exported by ginkgo.
The only functions that I haven't seen referenced in the code are:
```
FailNow()
Skipf(fomat string, args ...interface{})
SkipNow()
Skipped() bool
```
Do you want them removed ?

Reply:
I searched the `modules` folder for usages of `t.XXX` in non-test files (i.e., ignoring `xxx_test.go`) files, and the only usages I found were:

1. `t.Fatal()` / `t.Fatalf()`. These are by far the most common.
1. `t.Error()` / `t.Errorf()`. Only used a few times in goroutines because this method is goroutine-safe, unlike `t.Fatal`.
1. `t.Name()`: Used a few times.
1. `t.Logf()`: Used in just one place (`modules/ssh/agent.go`) and this is a bug. It should be replaced with `logger.Logf`. 

So, I think we can simplify this interface down to just 5 methods: two `Fatal` methods, two `Error` methods, and the `Name` method. If we want to be really minimalist, we could do just the `Fatalf` and `Errorf` method, as the ones without `f` are rarely used, and largely redundant. This much smaller interface would be far simpler to implement to integrate Terratest with other use cases!

Reply:
`Name()` method is not offered by `GinkgoT` interface
1. I can send a PR there to add it in the interface
2. Until that gets merged, if this interface requires .Name to exist, then it will not work with the current `gingko` versions

While this approach would be preferrable, it is highly likely that the feature will not be available (if at all) any time soon (given that it's a change at the API and might affect a lot of projects depending on it) 

these are the entries that make use of `t.Name()` in the `modules` dir not counting `_test.go` files:
```
docker/docker_compose.go:29:		Args:       append([]string{"--project-name", t.Name()}, args...),
logger/logger.go:47:	prefix := fmt.Sprintf("%s %s %s:", t.Name(), date.Format(time.RFC3339), CallerPrefix(callDepth+1))
test-structure/test_structure.go:75:	tmpRootFolder, err := files.CopyTerraformFolderToTemp(rootFolder, cleanName(t.Name()))
k8s/kubectl.go:114:	escapedTestName := url.PathEscape(t.Name())
```

Would you be open to the idea of replacing the use of `t.Name` in these files with something else ?

Reply:
Hm, `Name` is definitely important, especially for logging. What else would you propose replacing it with?

Reply:
I think you're right, there isn't an easy way around this. I think I can have a workaround outside of terratest, until ginkgo updates their interface.
I have updated the interface to contain only the methods you mentioned ( including `Name()` )